src/handlers.rs

```diff
@@ -673,3 +678,36 @@ fn handle_timer_clear(
   remove_timer(d, msg.id());
   Ok(null_buf())
 }
+
+fn handle_chmod_sync (
+  d: *const DenoC,
+  base: msg::Base,
+  _builder: &mut FlatBufferBuilder,
+) -> HandlerResult {
+  let msg = base.msg_as_chmod_sync().unwrap();
+  let path = msg.path().unwrap();
+  let mode = msg.mode().unwrap();
+  let deno = from_c(d);
+  debug!("handle_chmod_sync {} {}", path, mode);
+  if !deno.flags.allow_write {
+    let err = std::io::Error::new(
+      std::io::ErrorKind::PermissionDenied,
+      "allow_write is off.",
+    );
+    return Err(err.into());
+  }
+  let buf = CString::new(path).unwrap();
+  let _mode = CString::new(mode).unwrap();
+  let mut path_to_dynamic_lib = "";
+  if cfg!(target_os = "windows") {
+    path_to_dynamic_lib = "/home/srijan/Documents/chmod.dll"; // This will change to the path where deno will be present
+  } else {
+    path_to_dynamic_lib = "/home/srijan/Documents/chmod.so";  // This will change to the path where deno will be present 
```

Comment:
Hi Srijan - Thanks for looking into this but we shouldn't need any DLL loading to make this work. I think everything is in Rust's std library.

Reply:
@srijanreddy98 probably could try [std::fs::set_permissions](https://doc.rust-lang.org/std/fs/fn.set_permissions.html)

Reply:
@kevinkassimo I initially thought of using it but this method takes a permission struct which has only one method called set_readonly(x: bool). So you can only set permissions to -r--r--r-- or -rw-rw-rw-. So I was looking for other ways.

Reply:
@srijanreddy98 then probably [`PermissionExt`](https://doc.rust-lang.org/std/os/unix/fs/trait.PermissionsExt.html) is what you are looking for.
I believe [`OpenOptionsExt`](https://doc.rust-lang.org/std/os/unix/fs/trait.OpenOptionsExt.html) could also be used/

Reply:
@kevinkassimo will it work for windows?